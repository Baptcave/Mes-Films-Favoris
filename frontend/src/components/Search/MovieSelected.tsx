import styles from "../../styles/MovieSelected.module.css";
import { MovieFromIMDB } from "../../types/MovieFromIMDB";

type MovieSelectedProps = {
  movieSelected: MovieFromIMDB;
};

function MovieSelected({ movieSelected }: MovieSelectedProps) {
  const movie = movieSelected;
  return (
    <div className={styles.movieContainer}>
      <div className={styles.movieCard}>
        <div className={styles.imageContainer}>
          <img
            className={styles.image}
            src={`https://image.tmdb.org/t/p/w500${movie.poster_path}`}
            alt={`image de ${movie.title}`}
          />
        </div>
        <p>Titre : {movie.title}</p>
        <p>Date de sortie : {movie.release_date}</p>
        <p>Durée : {movie.runtime} minutes</p>
        <p>Note Imdb : {movie.vote_average} / 10</p>
        <p>Résumé : {movie.overview}</p>
      </div>
    </div>
  );
}

export default MovieSelected;
