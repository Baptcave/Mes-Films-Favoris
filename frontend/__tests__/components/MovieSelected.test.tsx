import "@testing-library/jest-dom";
import { render } from "@testing-library/react";
import MovieSelected from "@/components/Search/MovieSelected";

describe("MovieSelected", () => {
  it("renders correctly", () => {
    const res = render(
      <MovieSelected
        movieSelected={{
          id: "1",
          title: "Matrix",
          runtime: 135,
          release_date: "1999-03-31",
          vote_average: 8,
          poster_path: "/pEoqbqtLc4CcwDUDqxmEDSWpWTZ.jpg",
          overview:
            "Programmeur anonyme dans un service administratif le jour, Thomas Anderson devient Neo la nuit venue"
        }}
      />
    );

    expect(res.baseElement).toMatchSnapshot();
  });
});
