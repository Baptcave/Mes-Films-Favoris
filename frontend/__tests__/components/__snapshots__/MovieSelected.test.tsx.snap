// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MovieSelected renders correctly 1`] = `
<body>
  <div>
    <div>
      <div>
        <div>
          <img
            alt="image de Matrix"
            src="https://image.tmdb.org/t/p/w500/pEoqbqtLc4CcwDUDqxmEDSWpWTZ.jpg"
          />
        </div>
        <p>
          Titre : 
          Matrix
        </p>
        <p>
          Date de sortie : 
          1999-03-31
        </p>
        <p>
          Durée : 
          135
           minutes
        </p>
        <p>
          Note Imdb : 
          8
           / 10
        </p>
        <p>
          Résumé : 
          Programmeur anonyme dans un service administratif le jour, Thomas Anderson devient Neo la nuit venue
        </p>
      </div>
    </div>
  </div>
</body>
`;
